vscode-dotnet-runtime-extension/src/extension.ts

```diff
@@ -231,20 +249,29 @@ export function activate(vsCodeContext: vscode.ExtensionContext, extensionContex
                 return Promise.resolve(existingPath);
             }
 
-            const globalInstallerResolver = new GlobalInstallerResolver(sdkContext, commandContext.version);
+            if(!(await WebRequestWorker.isOnline(timeoutValue ?? defaultTimeoutValue, globalEventStream)))
+            {
+                const existingPathOffline = await worker.getSimilarExistingInstall(workerContext);
+                if(existingPathOffline?.dotnetPath)
+                {
+                    return Promise.resolve(existingPathOffline);
+                }

```

Comment:
This does not log message like the other?

Reply:
yikes it also doesnt set the architecture lemme refactor it out into a function

Reply:
good catch thank you!!